Updated Levonorgestrel model

mgsc1020 opened the pull request:
- Model updated to V12 and cleaned version for evaluation plan for healthy volunteers
- Model now uses peripheral venous blood (PVB) as output instead of venous blood (VB) for consistency across compound model assumptions when comparing to observed data
- Re-optimization of Lipophilicity, CLspec CYP3A4, and dissolution parameters when using PVB as main output

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive evaluation content: introduction, modeling strategy, data, parameters & assumptions, results & discussion, diagnostics, concentration‑time profiles, conclusion, references, input table, and title page.

* **New Features**
  * Added an automated qualification/report generation workflow to build and run evaluation reports.

* **Chores**
  * Added a detailed evaluation plan/configuration, updated README/title and description, and removed a tracked pointer to a large model asset.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Laura-Villain committed:
structure for evaluation plan

mgsc1020 committed:
Updated evaluation plan content sections

mgsc1020 committed:
Removed levonogestrel pksim and added updated levonogestrel json file

mgsc1020 committed:
Updated evaluation plan for levonogestrel

mgsc1020 committed:
Updated readme

mgsc1020 committed:
Updated readme

mgsc1020 committed:
Fix typos and enhance README content

Updated references and improved clarity in the README.

mgsc1020 committed:
Revise README for model details and version info

Updated README.md to reflect changes in model description and version information.

mgsc1020 committed:
Updated read me

Merge branch 'main' of https://github.com/Laura-Villain/Levonorgestrel

# Conflicts:
#	README.md

mgsc1020 committed:
Updated Levornogestrel model

mgsc1020 committed:
Updated read me

mgsc1020 committed:
Updated evaluation plan

mgsc1020 committed:
Updated workflow

mgsc1020 committed:
Updated evaluation plan content

mgsc1020 committed:
Updated evaluation plan content

mgsc1020 commented:
@nina-nauwelaerts can you review?

mgsc1020 committed:
Updated Json, removed unused BBs

mgsc1020 committed:
Updated model parameters and assumptions

nina-nauwelaerts commented:
- [ ] Please adapt the axis range/simulation time to the observed data

  - Cicali 2021_Levonorgestrel_IV_0.09 mg
  - Cicali 2021_Levonorgestrel_PO_0.03 mg
  - Natavio 2019_Levonorgestrel_PO_1.5 mg

- [ ] Please add observed data to graph for Cicali 2021_Levonorgestrel_PO_0.03 mg MD 28 days
- [ ] Is "_Levonorgestrel-CYP3A4-Parameter Identification Metabolite-Organism-Total fraction of dose-Levonorgestrel_" needed in Cicali 2021_Levonorgestrel_PO_0.09 mg
- [ ] Johansson 2002_Levonorgestrel_PO_1.5 mg has two Time Profiles with observed data. Please also link the data for both in the Observed Data sheet.

nina-nauwelaerts commented on README.md:
```diff
@@ -1,22 +1,22 @@
-Whole-body PBPK model of Levonorgestrel as well as ethinyl estradiol 
-
+# Levonorgestrel-Model
+Whole-body PBPK model of Levonorgestrel as well as ethinylestradiol 
```
Does the repository contain ethinylestradiol? I only saw levonorgestrel?

mgsc1020 replied:
yes just LNG, updated

nina-nauwelaerts commented on README.md:
```diff
@@ -1,22 +1,22 @@
+For further information on CYP3A4-mediated drug-drug interactions (DDI) when orally administered either alone or in combination with ethinylestradiol in healthy and obese women, refer to [[1]](https://ascpt.onlinelibrary.wiley.com/doi/full/10.1002/psp4.12572).
 
-The presented model represents an original model development project presented by Cicali *et al.* [1]. A complete summary of model development and application can be found in this manuscript and respective supplemental materials.
+### Version information
```
Do we not want to remove version information in the repository?

nina-nauwelaerts commented on README.md:
```diff
@@ -1,22 +1,22 @@
-
+# Levonorgestrel-Model
+Whole-body PBPK model of Levonorgestrel as well as ethinylestradiol 
 
```
Insert title according to template

```sugestion
## Repository files

nina-nauwelaerts commented on README.md:
```diff
@@ -1,22 +1,22 @@
-
-This levonogrestrel model is intended to be used as victim drug in CYP3A4-mediated drug-drug interactions (DDI) when orally administered either alone or in combination with ethinyl estradiol in healthy and obese women. 
+The present model represents an updated version of the model presented by Cicali *et al.* [[1]](https://ascpt.onlinelibrary.wiley.com/doi/full/10.1002/psp4.12572). 
+This levonorgestrel model incorporates metabolism by CYP3A4 and additional hepatic clearance. The current evaluation plan focuses on the performance of the levonorgestrel model in healthy women volunteers
```
```suggestion 
This levonorgestrel model incorporates metabolism by CYP3A4 and additional hepatic clearance. The current evaluation plan focuses on the performance of the levonorgestrel model in healthy female volunteers

nina-nauwelaerts commented on Evaluation/Input/Content/Section1_Introduction.md:
```diff
@@ -0,0 +1,10 @@
+LNG is a biopharmaceutics classification system (BCS) class I compound at the doses of interest, it is completely
+absorbed and demonstrates almost complete bioavailability ([Back 1981](#main-references)).LNG is cleared by cytochrome P450-3A4 (CYP3A4) as well as phase II enzymes, including uridine 5’-diphosphoglucuronosyltransferase or sulfotransferases. LNG is extensively bound to plasma proteins (~97.5%), primarily the sex hormone binding globulin (SHBG), and to less extent albumin ([Back 1981](#main-refere ...
+
+The herein presented model building and evaluation report evaluates the performance of the PBPK model for LNG in (healthy) women. The PBPK model of LNG has been developed and evaluated by comparing simulations to observed data of both intravenously and orally administered metformin covering a dosing range from 0.03 to 1.5 mg.
```
```suggestion 
The herein presented model building and evaluation report evaluates the performance of the PBPK model for LNG in (healthy) women. The PBPK model of LNG has been developed and evaluated by comparing simulations to observed data of both intravenously and orally (fasted/fed state) administered levonorgestrel covering a dosing range from 0.03 to 1.5 mg.

nina-nauwelaerts commented on Evaluation/Input/Content/Section1_Introduction.md:
```diff
@@ -0,0 +1,10 @@
+include progestin-only pills, subdermal implants, intrauterine devices, combined hormonal contraceptives, and emergency contraception ([Cicali 2021](#main-references)) . 
+
+LNG is a biopharmaceutics classification system (BCS) class I compound at the doses of interest, it is completely
+absorbed and demonstrates almost complete bioavailability ([Back 1981](#main-references)).LNG is cleared by cytochrome P450-3A4 (CYP3A4) as well as phase II enzymes, including uridine 5’-diphosphoglucuronosyltransferase or sulfotransferases. LNG is extensively bound to plasma proteins (~97.5%), primarily the sex hormone binding globulin (SHBG), and to less extent albumin ([Back 1981](#main-refere ...
```
```suggestion 
absorbed and demonstrates almost complete bioavailability ([Back 1981](#main-references)). LNG is cleared by cytochrome P450-3A4 (CYP3A4) as well as phase II enzymes, including uridine 5’-diphosphoglucuronosyltransferase or sulfotransferases. LNG is extensively bound to plasma proteins (~97.5%), primarily the sex hormone binding globulin (SHBG), and to less extent albumin ([Back 1981](#main-references)), ([FDA 2017](#main-references)).

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.1_Modeling_Strategy.md:
```diff
@@ -0,0 +1,20 @@
+
+The applied activity and variability of plasma proteins and active processes that are integrated into PK-Sim® are described in the publicly available PK-Sim® Ontogeny Database Version 7.3 ([Schlender 2016](#main-references)) or otherwise referenced for the specific process.
+
+The present model represents an updated version of the model presented by Cicali et al. [Cicali 2021](#main-references). A PBPK model for levonogestrel was developed and evaluated for a healthy adult volunteers for IV and oral administration. The mean PBPK model was developed using a standard female individual. The model was evaluated by comparing the pharmacokinetic simulation with observed in v ...
```
Was the model first developed for healthy adult volunteers (i.e. also male) or immediately for females only? These sentences are not very clear.

```suggestion 
The present model represents an updated version of the model presented by Cicali et al. ([Cicali 2021](#main-references)). A PBPK model for levonogestrel was developed and evaluated for healthy adult volunteers for IV and oral administration. The mean PBPK model was developed using a standard female individual. The model was evaluated by comparing the pharmacokinetic simulation with observed in vivo pharmacokinetic data taken from literature. The current levonogestrel model is based on the publication by Cicali et al [Cicali 2021](#main-references).

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.2_Data.md:
```diff
@@ -0,0 +1,51 @@
+| [Cicali 2021](#main-references) | 0.27 mg       |PO    |Healthy women receiving a single dose administration of lenovogestrel 
+
+
+The following studies were used for model evaluation:
```
The model evaluation data does not show as a table in the report.

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,48 @@
+### Absorption <a id="model-parameters-and-assumptions-absorption"></a>
+
+`logP` (as lipophilicity), `specific intestinal permeability`, and dissolution-related parameters were implemented with Lint80 or Weibull dissolution parameters depending on the formulation used in each study (`T80%` ot `T50%`, `dissolution shape`, `dissolution time` and/or `lag time`) were optimized by comparing model prediction with observed plasma PK data following oral administration.
```
The sentence doesn't seem to be correct?

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,48 @@
+
+### Automated Parameter Identification <a id="model-parameters-and-assumptions-parameter-identification"></a>
+
+This is the result of the final parameter identification.
```
Which data was used to fit the parameters? I could not find information about the parameter identification either here or in the model file.

mgsc1020 replied:
the datasets are separated into model building and verification

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,48 @@
+
+### Distribution <a id="model-parameters-and-assumptions-distribution"></a>
+
+The base disposition model developed with data after 0.09 mg IV levonogestrel administration was further expanded by incorporating aqueous solubility, intestinal permeability, and drug dissolution to simulate LNG exposure after oral administration.
```
Should this sentence be moved to Absorption section?

mgsc1020 replied:
Updated

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,48 @@
+
+The base disposition model developed with data after 0.09 mg IV levonogestrel administration was further expanded by incorporating aqueous solubility, intestinal permeability, and drug dissolution to simulate LNG exposure after oral administration.
+
+Unbound fraction (fu) was described with a specific binding process of levonogestrel to albumin and sex-hormone binding globulin (SHBG) as binding partners. Protein binding parameters (`Kd` and `Koff`) for albumin and SHBG were estimated by optimizing the model to subject-level data following IV administration of 0.09 mg of levonogestrel-only contraceptive to 18 women([Cicali 2021](#main-referenc ...
```
```suggestion 
Unbound fraction (fu) was described with a specific binding process of levonogestrel to albumin and sex-hormone binding globulin (SHBG) as binding partners. Protein binding parameters (`Kd` and `Koff`) for albumin and SHBG were estimated by optimizing the model to subject-level data following IV administration of 0.09 mg of levonogestrel-only contraceptive to 18 women ([Cicali 2021](#main-references)).

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,48 @@
+Unbound fraction (fu) was described with a specific binding process of levonogestrel to albumin and sex-hormone binding globulin (SHBG) as binding partners. Protein binding parameters (`Kd` and `Koff`) for albumin and SHBG were estimated by optimizing the model to subject-level data following IV administration of 0.09 mg of levonogestrel-only contraceptive to 18 women([Cicali 2021](#main-referenc ...
+
+Systemic levels of albumin and SHBG were based on the built-in expressed sequence tags database and on the clinical results
+by [Kuhnz 1994](#main-references) 
```
```suggestion 
by [Kuhnz 1994](#main-references).

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,48 @@
+
+The contribution of CYP3A4 to unbound levonogestrel hepatic
+clearance was back-calculated from a clinical DDI study
+comparing the systemic exposure of a combined LNG-EE
```
Is LNG-EE already defined before?

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,48 @@
+clearance was back-calculated from a clinical DDI study
+comparing the systemic exposure of a combined LNG-EE
+contraceptive administered with and without the strong
+CYP3A4 inhibitor telithromycin (i.e., net levonogestrel fraction metabolized by CYP3A4 when administered in combination with Ethinylestradiol). However, levonogestrel metabolic clearance was estimated to be ~ 30% lower for the Levonogestrel and Ethynilestradion combined hormonal therapy compared with levonogestrel-only formulations. Only a fraction of the levonogestrel metabolic pathway (70%) was ...
```
Do you want to specify the published DDI studies to which the fm was compared as references? 

```suggestion 
CYP3A4 inhibitor telithromycin (i.e., net levonogestrel fraction metabolized by CYP3A4 when administered in combination with Ethinylestradiol). However, levonogestrel metabolic clearance was estimated to be ~ 30% lower for the Levonogestrel and Ethynilestradiol combined hormonal therapy compared with levonogestrel-only formulations. Only a fraction of the levonogestrel metabolic pathway (70%) was subject to a subsequent interaction with a second strong CYP3A4 inhibitor, namely telithromycin ([Cicali 2021](#main-references)). Thus, the back calculated area under the concentration-time curve ratio (AUCR) resulting from the concomitant interaction on LNG exposure (i.e., eethynilestradiol and telithromycin) would be 2.1 (i.e., AUCR/0.7), which is equivalent to an levonogestrel fraction metabolized by CYP3A4 of 47%. Given the lack of mechanistic information, the current model lumps the contribution of phase II enzymes to levonogestrel into a nonspecific hepatic clearance ([Cicali 2021](#main-references)). The simulated levonogestrel fraction metabolized by CYP3A4 (fmCYP3A4) was verified by comparing to published clinical DDI studies. Relative tissue distribution of CYP3A4 implemented into the model was loaded from the 'PK-Sim® Ontogeny Database Version 7.3' (PK-Sim Ontogeny Database Version 7.3) using RT-PCR as data source.

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,48 @@
+comparing the systemic exposure of a combined LNG-EE
+contraceptive administered with and without the strong
+CYP3A4 inhibitor telithromycin (i.e., net levonogestrel fraction metabolized by CYP3A4 when administered in combination with Ethinylestradiol). However, levonogestrel metabolic clearance was estimated to be ~ 30% lower for the Levonogestrel and Ethynilestradion combined hormonal therapy compared with levonogestrel-only formulations. Only a fraction of the levonogestrel metabolic pathway (70%) was ...
+In addition to the CYP3A4 metabolism, total hepatic clearance was implented in the model as an additional elimination pathway
```
This is already specified above also so I would remove the sentence here (otherwise add a "." in the end).

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,48 @@
+contraceptive administered with and without the strong
+CYP3A4 inhibitor telithromycin (i.e., net levonogestrel fraction metabolized by CYP3A4 when administered in combination with Ethinylestradiol). However, levonogestrel metabolic clearance was estimated to be ~ 30% lower for the Levonogestrel and Ethynilestradion combined hormonal therapy compared with levonogestrel-only formulations. Only a fraction of the levonogestrel metabolic pathway (70%) was ...
+In addition to the CYP3A4 metabolism, total hepatic clearance was implented in the model as an additional elimination pathway
+Both `CYP3A4 CLspec/[enzyme]` and `Plasma Clearance` were optimized. 
```
Based on what data?

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,48 @@
+In addition to the CYP3A4 metabolism, total hepatic clearance was implented in the model as an additional elimination pathway
+Both `CYP3A4 CLspec/[enzyme]` and `Plasma Clearance` were optimized. 
+
+### Obserser for Levonogestrel concentrations <a id="model-parameters-and-assumptions-parameter-observer"></a>
```
```suggestion 
### Observer for Levonogestrel concentrations <a id="model-parameters-and-assumptions-parameter-observer"></a>

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,48 @@
+| `Dissolution T50%` |  
+| `Dissolution shape` |  
+| `Lag Time` |  
+
```
Accordign to final parameters secrtion 3.1.1.3.1.1 also plasma clearance?

nina-nauwelaerts commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,48 @@
+| `logP` | 
+| `CYP3A4 CLspec/[enzyme]` |     
+| `Specific Clearance` |      
+| `Kd` |         
```
Specify for ALB (i.e., not for SHBG?) for Kd and Koff

nina-nauwelaerts reviewed (commented):
@mgsc1020

mgsc1020 committed:
Update Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md

Co-authored-by: Nina Nauwelaerts <63354539+nina-nauwelaerts@users.noreply.github.com>

mgsc1020 committed:
Update README.md

Co-authored-by: Nina Nauwelaerts <63354539+nina-nauwelaerts@users.noreply.github.com>

mgsc1020 committed:
Update Evaluation/Input/Content/Section1_Introduction.md

Co-authored-by: Nina Nauwelaerts <63354539+nina-nauwelaerts@users.noreply.github.com>

mgsc1020 committed:
Update Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md

Co-authored-by: Nina Nauwelaerts <63354539+nina-nauwelaerts@users.noreply.github.com>

mgsc1020 committed:
Update Evaluation/Input/Content/Section1_Introduction.md

Co-authored-by: Nina Nauwelaerts <63354539+nina-nauwelaerts@users.noreply.github.com>

mgsc1020 committed:
Update Evaluation/Input/Content/Section2.1_Modeling_Strategy.md

Co-authored-by: Nina Nauwelaerts <63354539+nina-nauwelaerts@users.noreply.github.com>

mgsc1020 committed:
Update Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md

Co-authored-by: Nina Nauwelaerts <63354539+nina-nauwelaerts@users.noreply.github.com>

mgsc1020 committed:
Updated readme

mgsc1020 committed:
Updated Readme

mgsc1020 committed:
Updated Section2.3_Model_Parameters_and_Assumptions

mgsc1020 committed:
Updated Section2.3_Model_Parameters_and_Assumptions

mgsc1020 committed:
Updated Section2.3_Model_Parameters_and_Assumptions

mgsc1020 committed:
Updated Data dn Model Parameters and assumptions

mgsc1020 committed:
Updated Levonorgestrel-model.json

mgsc1020 committed:
Updated Levonorgestrel-model.json

mgsc1020 committed:
Updated model parameters and assumptions

mgsc1020 committed:
Updated model parameters and assumptions

mgsc1020 committed:
Update Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md

Co-authored-by: Nina Nauwelaerts <63354539+nina-nauwelaerts@users.noreply.github.com>

mgsc1020 committed:
Fix README punctuation and remove version info

Corrected punctuation in the model description and removed version information.

mgsc1020 commented:
https://github.com/Open-Systems-Pharmacology/Levonorgestrel/pull/2#issuecomment-3612420505
All addressed :)

nina-nauwelaerts commented:
Thanks @mgsc1020, just one last comment:

This figure is missing a title:
<img width="677" height="553" alt="image" src="https://github.com/user-attachments/assets/11831efa-66c5-4b90-b607-459f1ca6371e" />

mgsc1020 committed:
Updated Levonorgestrel.json

mgsc1020 commented:
@nina-nauwelaerts updated, final check?

Yuri05 commented:
@coderabbitai full review

Yuri05 commented:
@mgsc1020 Testing a new code reviewer AI :)
Please check the comments by **coderabbitai** above

mgsc1020 committed:
Update README.md

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

mgsc1020 committed:
Update Evaluation/workflow.R

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

mgsc1020 committed:
Updated readme

mgsc1020 committed:
Updated section 2.3

mgsc1020 committed:
Update Evaluation/Input/Content/Section2.2_Data.md

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

mgsc1020 commented on Evaluation/Input/Content/Section2.3_Model_Parameters_and_Assumptions.md:
```diff
@@ -0,0 +1,47 @@
+### Absorption <a id="model-parameters-and-assumptions-absorption"></a>
+
+LNG administration was further expanded by incorporating
+aqueous solubility, intestinal permeability, and drug dissolution
+to simulate LNG exposure after oral administration. The permeability of LNG (`specific intestinal permeability`), and dissolution-related parameters were optimized by comparing model prediction with observed plasma PK data following oral administration. Dissolution was implemented with Lint80 or Weibull dissolution parameters depending on the formulation used in each study (`T80%` ot `T50%`, `dis ...
```
abbreviation was defined

mgsc1020 committed:
Corrections to content

mgsc1020 committed:
Updated Section 2.2.

mgsc1020 commented on Evaluation/workflow.R:
```diff
@@ -0,0 +1,167 @@
+  #' Default is no watermark. `Label` objects from `tlf` package can be used to specify watermark font.
+  watermark <- ""
+
+  #' If not set, report created will be named `report.md` and located in the workflow folder namely `reOutputFolder`
+  reportFolder <- file.path(workingDirectory, "report")
+  reportPath <- file.path(reportFolder, "report.md")
+
```
Will maintain template code

mgsc1020 committed:
updated content

mgsc1020 commented:
@Yuri05 ready to merge? :D

mgsc1020 committed:
Updated spelling

Yuri05 commented:
@mgsc1020 merged :)
